lab_3/lab_3.md

```diff
@@ -498,3 +499,445 @@ def calculate_distance(unknown_profile: LanguageProfile,
+поле `self.n_gram_frequencies` соответствующими N-граммами и их частотами. 
+
+Если заполнение прошло успешно, метод возвращает `0`,
+в противном случае метод возвращает `1`.
+
+**Интерфейс**:
+```py
+class NGramTrie:
+  ...
+  def extract_n_grams_frequencies(self, n_grams_dictionary: dict) -> int:
+      pass
+```
+
+### Шаг 8.2. Открыть языковой профиль (Выполнение Шагов 7-8.2 соответствует 8 баллам)
+
+Теперь нам понадобится метод, который сможет открыть сохраненный
+прежде языковой профиль.
+
+Метод принимает на вход название файла `name` языкового профиля с расширением,
+например, `eng_profile.json`, который лежит в текущей директории. 
+
+Метод открывает профиль и заполняет соответствующие поля класса `LanguageProfile`.
+
+Метод возвращает `0`, если открытие прошло успешно и `1` в противном случае.
+
+- **Обратите внимание, что сохраненный профиль нужно адаптировать к классу `LanguageProfile`.**
+- **При открытии профиля, его название и количество N-грамм разных размерностей
+записывается в поля класса `self.language` и `self.n_words` соответственно.**
+- **Для корректного открытия поля `freq` необходимо разделить все представленные N-граммы на
+группы в соответствии с размерностью N-грамм, а затем создать под каждую размерность N-грамм
+экземпляр класса `NGramTrie` и заполнить его поле `self.n_gram_frequencies` соответствующими 
+N-граммами и их частотами, вызвав написанный до этого метод `extract_n_gram_frequencies`. 
+Все экземпляры класса `NGramTrie` затем помещаются в поле класса
+`LanguageProfile` `self.tries` как и у обычного профиля.**
+
+Например, при открытии профиля:
+```py
+{
+  "freq": {
+     "_p": 1, "am": 1, "g_": 1, "in": 1,
+    "gr": 1, "mi": 1, "mm": 1, "ng": 1,
+    "og": 1, "pr": 1, "ra": 1, "ro": 1,
+    "a": 1, "g": 2, "_": 2, "i": 1,
+    "m": 2, "n": 1, "o": 1, "p": 1, "r": 2
+  },
+  "n_words": [
+    9,
+    12
+  ],
+  "name": "en"
+}
+```
+Должны заполнится следующие поля:
+```py
+en_profile = LanguageProfile(letter_storage_object, 'en')
+print(en_profile.language)  # en
+print(en_profile.tries)  # [<__main__.NGramTrie object at 0x09B37E50>, <__main__.NGramTrie object at 0x09B37EB0>]
+print(en_profile.n_words)  # [69, 71]
+```
+Каждый экземпляр класса `NGramTrie` из поля `self.tries` должен
+иметь следующие униграммы и биграммы в полях `self.n_gram_frequencies`:
+```py
+print(new_en_profile.tries[0].n_gram_frequencies)  # {(1,): 2, (2,): 1, (3,): 2, (4,): 1, (5,): 2, (6,): 1, (7,): 2, (8,): 1, (9,): 1}
+print(new_en_profile.tries[1].n_gram_frequencies)  # {(1, 2): 1, (2, 3): 1, (3, 4): 1, (4, 5): 1, (5, 3): 1, (3, 6): 1, (6, 7): 1, (7, 7): 1, (7, 8): 1, (8, 9): 1, (9, 5): 1, (5, 1): 1}
+```
+
+Шаблон для открытия и чтения из файла с использованием модуля `json`:
+```py
+with open(name, 'r') as file:
+    profile = json.load(file)
+    print(profile['name'])  # 'en'
+    # your code goes here
+```
+
+**Интерфейс**:
+```py
+class LanguageProfile:
+  ...
+  def open(self, name: str) -> int:
+      pass
+```
```

Comment:
add task for start py: build profile from raw text -> save profile -> open profile -> detect + this is the source for test:

should be same:
build profile from raw text -> save profile -> open profile -> detect
build profile from raw text -> detect